### PR TITLE
refactor(samples): convert sample tests from ava to mocha

### PR DESCRIPTION
**Pull request by nareshqlogic**
Fixes convert all sample tests from ava to mocha [#2865](https://github.com/googleapis/google-cloud-node/issues/2865) (it's a good idea to open an issue first for discussion)

- [ ] Tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

### CODE REVIEW

File: samples/.eslintrc.yml [hunk 1]
```diff
@@ -1,3 +1,5 @@
 ---
+env:
```
Comment (JustinBeckwith) [inline samples/.eslintrc.yml]: We could move this down into `samples/system-test`

File: samples/package.json [hunk 2]
```diff
@@ -9,16 +9,15 @@
...
   },
   "dependencies": {
     "@google-cloud/text-to-speech": "^0.3.0",
     "async": "^2.5.0",
     "yargs": "^12.0.0"
   },
   "devDependencies": {
-    "@google-cloud/nodejs-repo-tools": "^3.0.0",
-    "ava": "^0.25.0"
+    "@google-cloud/nodejs-repo-tools": "^3.0.0"
```
Comment (JustinBeckwith) [inline samples/package.json]: Please add `mocha` here

File: samples/.eslintrc.yml (hunk 1 above)
Comment (nareshqlogic) [reply samples/.eslintrc.yml]: Sure , done

File: samples/package.json (hunk 2 above)
Comment (nareshqlogic) [reply samples/package.json]: Thank you Justin. I did the change.

**Comment by JustinBeckwith** [comment]
Looks like a few interesting test failures:

```

  7 passing (6s)
  3 failing
  1) should synthesize audio from ssml:
      AssertionError [ERR_ASSERTION]: true === false
      + expected - actual
      -true
      +false
      at Context.it (system-test/synthesize.test.js:54:10)
  2) should synthesize audio from text file:
      AssertionError [ERR_ASSERTION]: true === false
      + expected - actual
      -true
      +false
      at Context.it (system-test/synthesize.test.js:64:10)
  3) should synthesize audio from ssml file:
      AssertionError [ERR_ASSERTION]: true === false
      + expected - actual
      -true
      +false
      at Context.it (system-test/synthesize.test.js:74:10)
(node:108) [DEP0013] DeprecationWarning: Calling an asynchronous function without callback is deprecated.
(node:108) [DEP0013] DeprecationWarning: Calling an asynchronous function without callback is deprecated.
(node:108) [DEP0013] DeprecationWarning: Calling an asynchronous function without callback is deprecated.
(node:108) [DEP0013] DeprecationWarning: Calling an asynchronous function without callback is deprecated.
(node:108) [DEP0013] DeprecationWarning: Calling an asynchronous function without callback is deprecated.
```

**Comment by nareshqlogic** [comment]
Hi @JustinBeckwith ,
I think file unlink should be executed after each test case as it is creating same file(test-output.mp3) in each one test case. So I changed from after to afterEach block. Let me correct if I am wrong.

**Comment by JustinBeckwith** [comment]
Sounds good to me, but we probably want to make sure it actually exists first, or wrap the unlink in a try/catch :)

```

  1 passing (787ms)
  1 failing
  1) "after each" hook:
     Uncaught Error: ENOENT: no such file or directory, unlink 'test-output.mp3'
```

**Comment by nareshqlogic** [comment]
Thank you Justin. I warped the unlink in a try/catch.

File: samples/system-test/audioProfile.test.js [hunk 3]
```diff
@@ -33,10 +33,29 @@ const outputFile4 = 'watchAudioTest.mp3';
 before(tools.checkCredentials);
 
 after(async () => {
-  await fs.unlink(outputFile1);
-  await fs.unlink(outputFile2);
-  await fs.unlink(outputFile3);
-  await fs.unlink(outputFile4);
+  try {
```
Comment (JustinBeckwith) [inline samples/system-test/audioProfile.test.js]:
```
await Promise.all([outputFile1, outputFile2, outputFile3, outputFile4].map(fs.unlink).catch(e => {}))
```

Or something like that :)

Comment (nareshqlogic) [reply samples/system-test/audioProfile.test.js]: change done.

**Pull request merged**